core/storage/beam_job/gae_models.py

```diff
@@ -0,0 +1,150 @@
+# may be set by the Cloud Dataflow service, as a transition from
+# RUNNING. It may also be set via a Cloud Dataflow jobs.update call,
+# if the job has not yet reached a terminal state.
+BEAM_JOB_STATE_DONE = 'DONE'
+# The job has has failed. This is a terminal job state. This state may only be
+# set by the Cloud Dataflow service, and only as a transition from RUNNING.
+BEAM_JOB_STATE_FAILED = 'FAILED'
+# The job's run state isn't specified.
+BEAM_JOB_STATE_UNKNOWN = 'UNKNOWN'
+
+
+class BeamJobRunModel(base_models.BaseModel):
+    """Represents an individual execution of an Apache Beam job.
+
+    IDs are generated by Google Cloud Dataflow, not Oppia.
+    """
+
+    # The name of the job class that implements the job's logic.
+    job_name = datastore_services.StringProperty(required=True)
+    # The arguments provided to the job run.
+    job_arguments = datastore_services.StringProperty(repeated=True)
+    # The state of the job at the time the model was last updated.
+    latest_job_state = datastore_services.StringProperty(choices=[
+        BEAM_JOB_STATE_DONE, BEAM_JOB_STATE_FAILED,
+        BEAM_JOB_STATE_RUNNING, BEAM_JOB_STATE_PENDING,
+        BEAM_JOB_STATE_UPDATED, BEAM_JOB_STATE_STOPPED,
+        BEAM_JOB_STATE_CANCELLING, BEAM_JOB_STATE_CANCELLED,
+        BEAM_JOB_STATE_DRAINING, BEAM_JOB_STATE_DRAINED,
+        BEAM_JOB_STATE_UNKNOWN
+    ], required=True)
+
+    @property
+    def in_terminal_state(self):
+        """Returns whether the job run has reached a terminal state and will no
+        longer run.
+
+        Returns:
+            bool. Whether the job has reached a terminal state.
+        """
+        return self.latest_job_state in [
+            BEAM_JOB_STATE_CANCELLED, BEAM_JOB_STATE_DRAINED,
+            BEAM_JOB_STATE_UPDATED, BEAM_JOB_STATE_DONE, BEAM_JOB_STATE_FAILED
+        ]
+
+    @staticmethod
+    def get_deletion_policy():
+        """Model doesn't contain any data directly corresponding to a user."""
+        return base_models.DELETION_POLICY.NOT_APPLICABLE
+
+    @staticmethod
+    def get_model_association_to_user():
+        """Model does not contain user data."""
+        return base_models.MODEL_ASSOCIATION_TO_USER.NOT_CORRESPONDING_TO_USER
+
+    @classmethod
+    def get_export_policy(cls):
+        """Model doesn't contain any data directly corresponding to a user."""
+        return dict(super(BeamJobRunModel, cls).get_export_policy(), **{
+            'job_name': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'job_arguments': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'latest_job_state': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+        })
+
+
+class BeamJobRunResultModel(base_models.BaseModel):
+    """Represents the result of an Apache Beam job.
+
+    IDs are generated by Google Cloud Dataflow, not Oppia.
+    """
+
+    # The unordered text output generated by the corresponding Apache Beam job.
```

Comment:
Maybe explain in the comment why this needs to be a repeated property.

Reply:
Done.

Reply:
Oh -- I don't mean in that way, sorry for being unclear. I mean, why is it a list and not a string -- what does each element of the list represent?

(The current info might also be useful to include regardless, but please explain _why_ it makes the cost cheaper than the alternative.)

Reply:
Actually, I decided this is probably a case of premature optimization. Let's keep it simple for now and make it a single TextProperty, we can investigate making it bigger later if we need to.